src/agenda/Contacto.java

```diff
@@ -25,13 +19,46 @@ public Contacto(String nombre, String apellido, String email, String movil){
 		this.movil=movil;
 	}
 
-//Funcion para seteo de Nombre.
-	public void setNombre(String nomb){
-		this.nombre=nomb.toUpperCase();
+	public void setNombre(String nombre){
+		this.nombre=nombre.toUpperCase();
+	}
+	
+	public String getNombre() {
+		return nombre;
+	}
+	
+	public void setApellido(String apellido) {
+		this.apellido = apellido;
+	}
+	
+	public String getApellido() {
+		return apellido;
+	}
+	
+	public void setEmail(String email) {
+		this.email = email;
 	}
 	
 	public String getEmail(){
-		return this.email;
+		return email;
+	}
+
+	public void setMovil(String movil) {
+		this.movil = movil;
+	}
+
+	public String getMovil() {
+		return movil;
 	}
 	
+@Override
+    public boolean equals(Object obj) {
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        final Contacto other = (Contacto) obj;
+        boolean sameA = (this.email == other.email) || (this.email != null && this.email.equalsIgnoreCase(other.email));
```

Comment:
Si, entiendo, lo analice y verifique lo que mencionas, la corrección seria la siguiente, me confirmas que este bien? elimine las dos ultimas lineas y retorno sameA.

@Override
    public boolean equals(Object obj) {
        if (obj == null) return false;
        if (getClass() != obj.getClass()) return false;
        final Contacto other = (Contacto) obj;
        boolean sameA = (this.email == other.email) || (this.email != null && this.email.equalsIgnoreCase(other.email));
        return sameA;
    }